relayer/processor/types.go

```diff
@@ -310,3 +333,30 @@ func (c ChannelMessageCache) Merge(other ChannelMessageCache) {
 		c[k] = v
 	}
 }
+
+// IBCHeaderCache holds a mapping of IBCHeaders for their block height.
+type IBCHeaderCache map[uint64]provider.IBCHeader
+
+// Merge will merge another IBCHeaderCache into this one.
+func (c IBCHeaderCache) Merge(other IBCHeaderCache) {
+	for k, v := range other {
+		c[k] = v
+	}
+}
+
+// Prune will delete all map entries except for the most recent (keep).
+func (c IBCHeaderCache) Prune(keep int) {
+	keys := make([]uint64, len(c))
+	i := 0
+	for k := range c {
+		keys[i] = k
+		i++
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+	toRemove := len(keys) - keep - 1
+	if toRemove > 0 {
+		for i := 0; i < toRemove; i++ {
+			delete(c, keys[i])
```

Comment:
This looks subtle enough to deserve a unit test.

Reply:
+1

Reply:
added